### DISCUSSION TITLE
Unable to get 'Example App' Working

### DISCUSSION THREAD

**Discussion by geirman** [wheels-dev/wheels#1468] [Q&A] [answered]
For starters, I've been through all three of these resources...
  > https://www.forgebox.io/view/cfwheels-template-example-app
  > https://cfwheels.org/blog/getting-the-example-app-up-and-running/
  > https://github.com/cfwheels/cfwheels-example-app/wiki/Installation

I installed using the following commands...
```
wheels g app name=example datasourceName=exampleh2 template=cfwheels-template-example-app --setupH2
package install
server start
```

Under the cfwheels info tab is the following system info...
> CFML Engine: 	Lucee5.4.6.9
> CFWheels Version: 	2.5.0
> Java Runtime: 	OpenJDK Runtime Environment
> Java Version: 	22.0.2

# The Problem — Plugins Not Recognized
Initially, everything seems fine and the login page loads. But when I try logging in (`admin@domain.com/Password123!`), the following error results.
![Screenshot 2024-11-22 at 11 07 09 AM](https://github.com/user-attachments/assets/ae542d9c-4279-4fdb-b078-fcb1d16fa840)

Sure enough, the plugin seems to be missing b/c when I visit `/wheels/plugins/` I find this...
![2024-11-22 11 08 57](https://github.com/user-attachments/assets/8b02f32d-f9a6-4875-be06-bd8394feb07e)

Which is strange because these plugins do seem to be installed by evidence of the plugin folder...
![Screenshot 2024-11-22 at 11 15 57 AM](https://github.com/user-attachments/assets/c5961646-2d4d-4fcb-b786-2b6d905a8a6d)

If something doesn't work, try turning it off then back on again...right? But, If I reload the framework (`/login?reload=true&password=please`), everything in the plugins directory is deleted forcing me to run `install` again. Restarting the server has no impact on whether the plugins are recognized either. 

So, I'm stuck. 

In CFWheels 1.4.6, which I'm using for another project, the plugins are installed into the directory as a zip file. Deleting the zip will delete the plugin when the server is initialized. Apparently, I'm not sure how to persist the plugin in version 2.5 nor am I understanding why the plugins aren't being recognized. Help me please 🙏

**Comment by neokoenig** [comment, marked as answer]
Do `set(deletePluginDirectories=false)` to remove the plugin deleting folder behaviour. The zip file should persist though?

**Reply by geirman** [reply]
Thanks Tom! This fixed the issue so now I'm able to login. And when I visit the `/wheels/plugins/` page, they now show up as expected.

**Reply by geirman** [reply]
I do however get a warning, but at least I'm no longer blocked...
![Screenshot 2024-11-22 at 12 26 51 PM](https://github.com/user-attachments/assets/94353282-1c57-4ece-bbee-2abcc68b0744)

**Reply by neokoenig** [reply]
Yeah don't worry about that - shouldn't cause any issues.

**Comment by neokoenig** [comment]
The more likely explanation is the path to the plugins folder is incorrect.
